src/config/ConfigManager.cpp

```diff
@@ -21,22 +21,28 @@ CConfigManager::CConfigManager() {
     setDefaultVars();
     setDefaultAnimationVars();
 
-    std::string CONFIGPATH;
-    if (g_pCompositor->explicitConfigPath == "") {
-        static const char* const ENVHOME = getenv("HOME");
-        CONFIGPATH                       = ENVHOME + (ISDEBUG ? (std::string) "/.config/hypr/hyprlandd.conf" : (std::string) "/.config/hypr/hyprland.conf");
-    } else {
-        CONFIGPATH = g_pCompositor->explicitConfigPath;
-    }
-
-    configPaths.emplace_back(CONFIGPATH);
+    configPaths.emplace_back(getMainConfigPath());
 
     Debug::disableLogs = &configValues["debug:disable_logs"].intValue;
     Debug::disableTime = &configValues["debug:disable_time"].intValue;
 
     populateEnvironment();
 }
 
+std::string CConfigManager::getMainConfigPath() {
+    if (g_pCompositor->explicitConfigPath != "")
+        return g_pCompositor->explicitConfigPath;
+    
+    std::string config_home = std::string(getenv("XDG_CONFIG_HOME"));
```

Comment:
we use camelCase

+ `std::string(const char*)` requires a nullcheck for the char*

Reply:
Using std::string in that case makes it harder to check if getenv returns nullptr.